python/lsst/ap/association/assoc_afw_utils.py

```diff
@@ -0,0 +1,279 @@
+
+
+def get_ccd_visit_info_from_exposure(exposure):
+    """
+    Extract info on the ccd and visit from the exposure.
+
+    Paramters
+    ---------
+    exposure : `lsst.afw.image.Exposure`
+        Exposure to store information from.
+
+    Returns
+    -------
+    values : `dict` of ``values``
+        List values representing info taken from the exposure.
+    """
+    visit_info = exposure.getInfo().getVisitInfo()
+    sphPoint = exposure.getWcs().getSkyOrigin()
+    flux0, flux0_err = exposure.getCalib().getFluxMag0()
+    filter_obj = exposure.getFilter()
+    # Values list is:
+    # [CcdVisitId ``int``,
+    #  ccdNum ``int``,
+    #  filterName ``str``,
+    #  RA WCS center ``degrees``,
+    #  DEC WCS center ``degrees``,
+    #  exposure time ``seconds``,
+    #  dateTimeMJD ``seconds``,
+    #  flux zero point ``counts``,
+    #  flux zero point error ``counts``]
+    values = {'ccdVisitId': visit_info.getExposureId(),
+              'ccdNum': exposure.getDetector().getId(),
+              'filterName': filter_obj.getName(),
+              'filterId': filter_obj.getId(),
+              'ra': sphPoint.getRa().asDegrees(),
+              'decl': sphPoint.getDec().asDegrees(),
+              'expTime': visit_info.getExposureTime(),
+              'expMidptMJD': visit_info.getDate().nsecs() * 10 ** -9,
+              'fluxZeroPoint': flux0,
+              'fluxZeroPointErr': flux0_err}
+    return values
+
+
+def add_dia_source_aliases_to_catalog(source_catalog):
+    """Create the needed aliases on fluxes and other columns if the input
+    catalog does not already have them.
+
+    Parameters
+    ----------
+    source_catalog : `lsst.afw.table.SourceCatalog`
+        Input source catalog to add alias columns to.
+    """
+    if not source_catalog.getSchema().contains(make_minimal_dia_source_schema()):
+        # Create aliases to appropriate flux fields if they exist.
+        schema_names = source_catalog.getSchema().getNames()
+        if 'base_PsfFlux_flux' in schema_names and \
+           'base_PsfFlux_fluxSigma' in schema_names and \
+           'psFlux' not in schema_names and \
+           'psFluxErr' not in schema_names:
+                alias_map = source_catalog.getSchema().getAliasMap()
+                alias_map.set('psFlux', 'base_PsfFlux_flux')
+                alias_map.set('psFluxErr', 'base_PsfFlux_fluxSigma')
+
+
+def convert_dia_source_to_asssoc_schema_and_calibrate(dia_sources,
+                                                      obj_ids=None,
+                                                      exposure=None):
+    """Create aliases in for the input source catalog schema and overwrite
+    values if requested.
+
+    Parameters
+    ----------
+    dia_sources : `lsst.afw.table.SourceCatalog`
+        Input source catalog to alias and overwrite values of.
+    obj_ids:  array-like of `int`
+        DIAObject ids to associated with the input DIASources. Overwrites
```

Comment:
"to associate with"

Also, note that if not None, it should correspond 1:1 with `dia_sources`.